nb_conda_kernels/install.py

```diff
@@ -20,67 +56,50 @@ def install(enable=False, disable=False, prefix=None):
+            os.makedirs(path)
+
+    cm = BaseJSONConfigManager(config_dir=path)
+    cfg = cm.get("jupyter_notebook_config")
+
+    log.debug("Existing config in {}...\n{}".format(path, pretty(cfg)))
+
+    nb_app = cfg.setdefault("NotebookApp", {})
 
     if enable:
-        if prefix is not None:
-            path = join(prefix, "etc", "jupyter")
-            if not exists(path):
-                print("Making directory", path)
-                os.makedirs(path)
-        else:
-            path = jupyter_config_dir()
-
-        cm = ConfigManager(config_dir=path)
-        print("Enabling nb_conda_kernels in", cm.config_dir)
-        cfg = cm.get("jupyter_notebook_config")
-        print("Existing config...")
-        pprint(cfg)
-
-        notebook_app = (cfg.setdefault("NotebookApp", {}))
-        if "kernel_spec_manager_class" not in notebook_app:
-            cfg["NotebookApp"]["kernel_spec_manager_class"] = "nb_conda_kernels.CondaKernelSpecManager"
-
-        cm.update("jupyter_notebook_config", cfg)
-        print("New config...")
-        pprint(cm.get("jupyter_notebook_config"))
-
-    if disable:
-        if prefix is not None:
-            path = join(prefix, "etc", "jupyter")
-        else:
-            path = jupyter_config_dir()
-
-        cm = ConfigManager(config_dir=path)
-        print("Disabling nb_conda_kernels in", cm.config_dir)
-        cfg = cm.get("jupyter_notebook_config")
-        print("Existing config...")
-        pprint(cfg)
-
-        kernel_spec_manager = cfg["NotebookApp"]["kernel_spec_manager_class"]
-
-        if "nb_conda_kernels.CondaKernelSpecManager" == kernel_spec_manager:
-            cfg["NotebookApp"].pop("kernel_spec_manager_class")
-
-        cm.set("jupyter_notebook_config", cfg)
-        print("New config...")
-        pprint(cm.get("jupyter_notebook_config"))
+        nb_app.update({KSMC: CKSM})
+    elif disable and nb_app.get(KSMC, None) == CKSM:
+        nb_app.pop(KSMC)
 
-if __name__ == '__main__':
+    log.debug("Writing config in {}...".format(path))
+
+    cm.set("jupyter_notebook_config", cfg)
+
+    cfg = cm.get("jupyter_notebook_config")
 
-    parser = argparse.ArgumentParser(
-        description="Installs nbextension")
-    parser.add_argument(
-        "-e", "--enable",
-        help="Automatically load nb_conda_kernels on notebook launch",
-        action="store_true")
-    parser.add_argument(
-        "-d", "--disable",
-        help="Remove nb_conda_kernels from config on notebook launch",
-        action="store_true")
-    parser.add_argument(
-        "-p", "--prefix",
-        help="prefix where to load nb_conda_kernels config",
-        action="store")
+    log.debug("Verifying config in {}...\n{}".format(path, pretty(cfg)))
 
+    if enable:
+        assert cfg["NotebookApp"][KSMC] == CKSM
+    else:
+        assert KSMC not in cfg["NotebookApp"]
+
+    log.info("{}abled nb_conda_kernels".format("En" if enable else "Dis"))
+
```

Comment:
Yeah, sorry about the really-similar magic strings, but i was having trouble figuring out where stuff was going on. I think it's pretty readable now...